.github/scripts/auto-backport.py

```diff
@@ -0,0 +1,182 @@
+                except GitCommandError as e:
+                    logging.warning(f'Cherry-pick conflict on commit {commit}: {e}')
+                    remote = 'fork'
+                    is_draft = True
+                    repo_local.git.add(A=True)
+                    repo_local.git.cherry_pick('--continue')
+            repo_local.git.push(remote, new_branch_name, force=True)
+            create_pull_request(repo, new_branch_name, backport_base_branch, pr, backport_pr_title, commits,
+                                is_draft=is_draft)
+        except GitCommandError as e:
+            logging.warning(f"GitCommandError: {e}")
+
+
+def create_pr_comment_and_remove_label(pr):
+    comment_body = f':warning:  @{pr.user.login} PR body does not contain a valid reference to an issue '
+    comment_body += ' based on [linking-a-pull-request-to-an-issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/using-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword)'
+    comment_body += ' and can not be backported\n\n'
+    comment_body += 'The following labels were removed:\n'
+    labels = pr.get_labels()
+    pattern = re.compile(r"backport/\d+\.\d+$")
+    for label in labels:
+        if pattern.match(label.name):
+            print(f"Removing label: {label.name}")
+            comment_body += f'- {label.name}\n'
+            pr.remove_from_labels(label)
+    comment_body += f'\nPlease add the relevant backport labels after PR body is fixed'
+    pr.create_issue_comment(comment_body)
+
+
+def main():
+    args = parse_args()
+    base_branch = args.base_branch.split('/')[2]
+    promoted_label = 'promoted-to-master'
+    repo_name = args.repo
+    if 'scylla-enterprise' in args.repo:
+        promoted_label = 'promoted-to-enterprise'
+    stable_branch = base_branch
+    backport_branch = 'branch-'
+
+    backport_label_pattern = re.compile(r'backport/\d+\.\d+$')
+
+    g = Github(github_token)
+    repo = g.get_repo(repo_name)
+    user = g.get_user()
+    closed_prs = []
+    start_commit = None
+
+    if args.commits:
+        start_commit, end_commit = args.commits.split('..')
+        commits = repo.compare(start_commit, end_commit).commits
+        for commit in commits:
+            for pr in commit.get_pulls():
+                closed_prs.append(pr)
+    if args.pull_request:
+        start_commit = args.head_commit
+        pr = repo.get_pull(args.pull_request)
+        closed_prs = [pr]
+
+    for pr in closed_prs:
+        labels = [label.name for label in pr.labels]
+        backport_labels = [label for label in labels if backport_label_pattern.match(label)]
+        if promoted_label not in labels:
+            print(f'no {promoted_label} label: {pr.number}')
+            continue
+        if not backport_labels:
+            print(f'no backport label: {pr.number}')
+            continue
+        commits = get_pr_commits(repo, pr, stable_branch, start_commit)
```

Comment:
do you plan to port all the follw-up changes to this repo? i found that at least https://github.com/scylladb/scylla-pkg/commit/bcd30e973fef25b2c50d1f2ea5dc027fb919bb66 is not included? is this on purpose?

Reply:
The patch you mention is the only one I removed (it also doesn't require a backport since it's relevant for `master`). I removed it because it's related to https://github.com/scylladb/scylladb/issues/20021 and there is a request for adding validation in `pull_github_pr.sh` as well. So I will implement it together in a follow-up patch